test/e2e/statefulset_arbitrary_config/statefulset_arbitrary_config_test.go

```diff
@@ -34,13 +34,17 @@ func TestStatefulSetArbitraryConfig(t *testing.T) {
 
 	mdb.Spec.StatefulSetConfiguration.SpecWrapper.Spec.Template.Spec.Containers[1].ReadinessProbe = &corev1.Probe{TimeoutSeconds: 100}
 
+	serviceName := "database"
+	mdb.Spec.StatefulSetConfiguration.SpecWrapper.Spec.ServiceName = serviceName
+
 	tester, err := mongotester.FromResource(t, mdb)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	t.Run("Create MongoDB Resource", mongodbtests.CreateMongoDBResource(&mdb, ctx))
 	t.Run("Basic tests", mongodbtests.BasicFunctionality(&mdb))
+	t.Run("Test setting Service Name", mongodbtests.ServiceWithNameExists(&mdb, serviceName))
```

Comment:
we are now **only** testing the service creation when it is specified in the StatefulSetSpec override, we could add this to the `BasicTests` tests which are run in every test to make sure the service name is as expected when not overriding through sts override. WDYT?

Reply:
do you mean adding a call to ServiceWithNameExists to BasicFunctionality and call it with m.Name + "-svc"?